# Deskripsi Data

In [1]:
import pandas as pd
import numpy as np
import mysql.connector as mysqlconnector
import psycopg2 as pgconnector

### Contoh Data

In [48]:
# Koneksi database postgresql

pgconn = pgconnector.connect("dbname=iris_db user=postgres password=zanra2401 host=localhost port=2005")

pgcursor = pgconn.cursor()

In [49]:
# Koneksi Mysql

myconn = mysqlconnector.connect(
    host="localhost",
    port=3306,
    user="root",
    password="",
    database="iris_db"
)

mycursor = myconn.cursor()


In [ ]:
# Mengambil data dari database postgresql

pgcursor.execute("SELECT petal_length, petal_width, species FROM iris_table")
data_postgre = pgcursor.fetchall()

In [ ]:
# Mengambil data dari databse MySql

mycursor.execute("SELECT sepal_length, sepal_width FROM iris_table")
data_mysql = mycursor.fetchall()

In [52]:
mycursor.close()
pgcursor.close()
myconn.close()
pgconn.close()

In [55]:
# Gabung Data

sepal = np.array(data_mysql)
petal_species = np.array(data_postgre)

iris = np.hstack((sepal, petal_species))

In [56]:
# Iris DataFrame

iris_dataframe = pd.DataFrame(iris, columns=['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'species'])
iris_dataframe

sepal_length sepal_width petal_length petal_width         species
0           5.10        3.50         1.40        0.20     Iris-setosa
1           4.90        3.00         1.40        0.20     Iris-setosa
2           4.70        3.20         1.30        0.20     Iris-setosa
3           4.60        3.10         1.50        0.20     Iris-setosa
4           5.00        3.60         1.40        0.20     Iris-setosa
..           ...         ...          ...         ...             ...
145         6.70        3.00         5.20        2.30  Iris-virginica
146         6.30        2.50         5.00        1.90  Iris-virginica
147         6.50        3.00         5.20        2.00  Iris-virginica
148         6.20        3.40         5.40        2.30  Iris-virginica
149         5.90        3.00         5.10        1.80  Iris-virginica

[150 rows x 5 columns]

### Jumlah Baris, Kolom, Nama Atribut dan Tipe Data

In [57]:
print("Baris: ", iris_dataframe.shape[0])
print("Kolom: ", iris_dataframe.shape[1])

Baris:  150
Kolom:  5


Data dari dataset iris tersebut memiliki jumlah baris dan kolom sebagai berikut:  

Baris: 150 baris  
Kolom: 5 kolom  

Data tersebut memiliki Atribut memiliki dengan nama sebagai berikut:  
1. sepal_lenght : Tipe Data merupakan Numerik (Continous) 
2. sepal_width : Tipe Data merupakan Numerik (Continous) 
3. petal_lenght : Tipe Data merupakan Numerik (Continous) 
4. petal_width : Tipe Data merupakan Numerik (Continous) 
5. species : Tipe Data Merupakan Kategorikal

